### ISSUE TITLE
Unable to reproduce

### ISSUE THREAD

**Issue by MNikdan** [hamishivi/automated-instruction-selection#34]
Hi.

I am trying to reproduce the single-task RDS+ results (with a pool of 200k samples) from the paper. I run everything with three seeds, but I consistently get slightly lower results compared to the numbers reported in the paper. Here are the numbers I get:

| Method   | mmlu          | gsm           | bbh           | tydiqa        | codex         | squad         | Average       |
|:---------|:--------------|:--------------|:--------------|:--------------|:--------------|:--------------|:--------------|
| RDS+     | 45.2 ± 0.5536 | 18.6 ± 0.6463 | 42.4 ± 0.4460 | 51.6 ± 0.9293 | 26.8 ± 1.1484 | 87.8 ± 0.1829 | 45.4 ± 0.6511 |

Here's how I'm running the experiments:

```
SEED=1
MODEL=meta-llama/Llama-2-7b-hf
DATA_PATH=/path/to/training_data

# create the index
python -m minimal_multitask.compute_influence_cosinesim \
    --model_name ${MODEL} \
    --seed ${SEED} \
    --train_dataset ${DATA_PATH}/subset200k_seed${SEED}.jsonl \
    --eval_dataset alpacafarm \
    --index_path ${DATA_PATH}/subset200k_seed${SEED}_rdsp_index.pt \
    --save_dir tmp/ \
    --batch_size 1 \
    --pooling weighted_mean


# calculate the influence scores
for dataset in gsm8k_shots bbh_shots tydiqa_shots codex squad mmlu_shots; do
    python -m minimal_multitask.compute_influence_cosinesim \
        --model_name ${MODEL} \
        --seed ${SEED} \
        --train_dataset ${DATA_PATH}/subset200k_seed${SEED}.jsonl \
        --eval_dataset $dataset \
        --index_path ${DATA_PATH}/subset200k_seed${SEED}_rdsp_index.pt \
        --save_dir ${DATA_PATH}/subset200k_infs_seed${SEED}/ \
        --batch_size 1 \
        --pooling weighted_mean
done


# perform the selection
for dataset in gsm8k_shots bbh_shots tydiqa_shots codex squad mmlu_shots; do
    python -m minimal_multitask.get_top_influences \
        --input_files ${DATA_PATH}/subset200k_infs_seed${SEED}/${dataset}_cossim.pkl \
        --output_file ${DATA_PATH}/subset200k_selections_seed${SEED}/${dataset}_top10k.json \
        --output_size 10000 --selection_method max \
        --train_datasets ${DATA_PATH}/subset200k_seed${SEED}.jsonl \
        --output_dataset
done


# run the training
CKPT_PATH=/path/to/checkpoints/dir
for dataset in gsm8k_shots bbh_shots tydiqa_shots codex squad mmlu_shots; do
    RUN_NAME=rdsp_tulu_v2_${dataset}_200k_top10k_seed${SEED}

    python -m minimal_multitask.instruction_tune \
        --model_name ${MODEL} \
        --output_dir ${CKPT_PATH}/${RUN_NAME} \
        --per_device_train_batch_size 1 \
        --gradient_accumulation_steps 128 \
        --num_train_epochs 2 \
        --learning_rate 2e-5 \
        --seed ${SEED} \
        --warmup_ratio 0.03 \
        --lr_scheduler_type linear \
        --weight_decay 0. \
        --evaluation_strategy no \
        --save_strategy no \
        --logging_steps 1 \
        --is_llama=True \
        --use_hf_auth_token True \
        --train_dataset "${DATA_PATH}/subset200k_selections_seed${SEED}/${dataset}_top10k.json"
done
```

And finally I use the `eval.sh` provided to get the final accuracy/F1 numbers. Can you please take a look and see whether I'm doing something wrong?

Thanks in advance.

**Comment by hamishivi** [comment]
Hi, the commands look correct at a glance. There may be some extra variance based on the subset you use, I uploaded the one (I am pretty sure) that I used here: https://huggingface.co/datasets/hamishivi/200k-tulu-2-unbalanced

**Comment by MNikdan** [comment]
Thanks a lot! I'll give it a try.

**Comment by MNikdan** [comment]
OK, indeed with this subset, the numbers are much closer to the ones from the paper.

| Method   | mmlu          | gsm           | bbh           | tydiqa        | codex         | squad         | Average       |
|:---------|:--------------|:--------------|:--------------|:--------------|:--------------|:--------------|:--------------|
| RDS+     | 46.4 | 23.7 | 42.8 | 50.9 | 31.8 | 86.3 | 47.0 |

I assume now the remaining difference simply comes from the training seed.
